python/semantic_kernel/data/text_search/vector_store_text_search.py

```diff
@@ -0,0 +1,191 @@
+# Copyright (c) Microsoft. All rights reserved.
+
+from collections.abc import AsyncIterable, Callable
+from typing import TYPE_CHECKING, Any, Generic, TypeVar
+
+from pydantic import model_validator
+
+from semantic_kernel.connectors.ai.embeddings.embedding_generator_base import EmbeddingGeneratorBase
+from semantic_kernel.data.kernel_search_results import KernelSearchResults
+from semantic_kernel.data.text_search.text_search import TextSearch
+from semantic_kernel.data.text_search.text_search_result import TextSearchResult
+from semantic_kernel.data.vector_search.vector_search_options import VectorSearchOptions
+from semantic_kernel.data.vector_search.vector_search_result import VectorSearchResult
+from semantic_kernel.data.vector_search.vector_text_search import VectorTextSearchMixin
+from semantic_kernel.data.vector_search.vectorizable_text_search import VectorizableTextSearchMixin
+from semantic_kernel.data.vector_search.vectorized_search import VectorizedSearchMixin
+from semantic_kernel.kernel_pydantic import KernelBaseModel
+
+if TYPE_CHECKING:
+    from semantic_kernel.data.search_options import SearchOptions
+
+TModel = TypeVar("TModel")
+_T = TypeVar("_T")
+
+
+class VectorStoreTextSearch(KernelBaseModel, TextSearch, Generic[TModel]):
+    """Class that wraps a Vector Store Record Collection to expose as a Text Search.
+
+    Preferably the class methods are used to create an instance of this class.
+    Otherwise the search executes in the following order depending on which store was set:
+    1. vectorizable_text_search
+    2. vector_text_search
+    3. vectorized_search (after calling the embedding service)
+
+    Args:
+        vectorizable_text_search: A vector store record collection with a method to search for vectorizable text.
+        vectorized_search: A vector store record collection with a method to search for vectors.
+        vector_text_search: A vector store record collection with a method to search for text.
+        embedding_service: An embedding service to use for vectorized search.
+        string_mapper: A function to map a record to a string.
+        text_search_results_mapper: A function to map a record to a TextSearchResult.
+
+    """
+
+    vectorizable_text_search: VectorizableTextSearchMixin | None = None
+    vectorized_search: VectorizedSearchMixin | None = None
+    vector_text_search: VectorTextSearchMixin | None = None
+    embedding_service: EmbeddingGeneratorBase | None = None
+    string_mapper: Callable[[TModel], str] | None = None
+    text_search_results_mapper: Callable[[TModel], TextSearchResult] | None = None
+
+    @model_validator(mode="before")
+    @classmethod
+    def _validate_stores(cls, data: dict[str, Any]) -> dict[str, Any]:
+        """Validate the capabilities."""
+        if (
+            not data.get("vectorizable_text_search")
+            and not data.get("vectorized_search")
+            and not data.get("vector_text_search")
+        ):
+            raise ValueError(
```

Comment:
We've added so many "custom" exceptions in SK. At what point do we stick with the "generic" ValueError versus adding a custom "TextSearchException"? Or something similar...